tools/modules.php

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env php
+<?php
+/**
+ * The modules.php script a database and automatically adds any new modules from
+ * either the project/modules or loris/modules directory and maintains the
+ * modules table based on the filesystem state.
+ *
+ * If the --add parameter is provided, it will add any new directories that
+ * have a valid \Module descriptor to the modules table.
+ *
+ * If the --remove parameter is provided, it will remove anything in the
+ * modules table that can not be instantiated with Module::factory from
+ * the modules table.
+ *
+ * If the -n flag is provided, it will not actually add/delete from the
+ * table, but only tell you what it would otherwise do.
+ *
+ * PHP Version 7
```

Comment:
Can you add the usual author, license etc here? 

It'll be needed soon anyway: #5766

Reply:
done